### ISSUE TITLE
[FEATURE] Add support for other ArgoCD application types (e.g. `kustomize`)

### ISSUE THREAD

**Issue by spkane** [aws-ia/terraform-aws-eks-blueprints#621]
#### Is your feature request related to a problem? Please describe
Targeting Helm by default is understandable for the feature set and flexibility that a project like this needs to provide, but it can make things overly complex due to the multiple layers of indirection, especially when it is not always required.

It would be really nice to be able to add ArgoCD applications via this mechanism that utilize other tools that ArgoCD natively supports, like `kustomize`.

#### Describe the solution you'd like

I'd love to be able to define a block like the following, that defines the `type` of application (helm maybe being the default) and have it work.

```hcl
  argocd_applications = {
    ...
    "internal-workloads-${var.zone}" = {
      path     = "workloads/envs/overlays/${var.zone}"
      repo_url = local.addon_repos.internal_addons.url
      project  = "default"
      type = "kustomize"
    }
  }
```

Currently, this will result in an error:
`Chart.yaml: no such file or directory`
because of the way the ArgoCD application is defined (as being a Helm-based application)

#### Describe alternatives you've considered

I can just wire this manifest in myself, but then there are just more ways to install things. If one is to go this route, then it would be nice to use this to seed all the initial apps, but use the underlying tool that keeps the implementation as simple as possible for the usecase.

#### Additional context

I could probably help write up a PR, if there is interest.

**Comment by askulkarni2** [comment]
Hi @spkane, great call out! If you want to contribute this feature, we would definitely welcome the PR.

**Comment by spkane** [comment]
I am going to take a stab at this afternoon.

I'll probably try and use a simple manifest apply for the customize-based Argo Application manifests, but it is worth noting that this also exists:

https://github.com/oboukili/terraform-provider-argocd

**Comment by bryantbiggs** [comment]
Closed in #640 - thank you again @spkane !
